lib/ReactViews/DragDropFile.tsx

```diff
@@ -97,72 +126,79 @@ class DragDropFile extends Component<PropsType> {
     e.preventDefault();
     e.stopPropagation();
     e.dataTransfer.dropEffect = "copy";
-    this.target = e.target;
-  }
+    targetRef.current = e.target;
+  });
 
-  handleDragOver(e: DragEvent) {
+  const handleDragOver = (e: DragEvent) => {
     e.preventDefault();
-  }
+  };
 
-  @action
-  handleDragLeave(e: MouseEvent) {
+  const handleDragLeave = action((e: MouseEvent) => {
     e.preventDefault();
     if (e.screenX === 0 && e.screenY === 0) {
-      this.props.viewState.isDraggingDroppingFile = false;
+      viewState.isDraggingDroppingFile = false;
     }
-    if (e.target === document || e.target === this.target) {
-      this.props.viewState.isDraggingDroppingFile = false;
+    if (e.target === document || e.target === targetRef.current) {
+      viewState.isDraggingDroppingFile = false;
     }
-  }
-
-  @action
-  handleMouseLeave() {
-    this.props.viewState.isDraggingDroppingFile = false;
-  }
-
-  render() {
-    return (
-      <div
-        onDrop={this.handleDrop.bind(this)}
-        onDragEnter={this.handleDragEnter.bind(this)}
-        onDragOver={this.handleDragOver.bind(this)}
-        onDragLeave={this.handleDragLeave.bind(this)}
-        onMouseLeave={this.handleMouseLeave.bind(this)}
-        className={classNames(Styles.dropZone, {
-          [Styles.isActive]: this.props.viewState.isDraggingDroppingFile
-        })}
+  });
+
+  const handleMouseLeave = action(() => {
+    viewState.isDraggingDroppingFile = false;
+  });
+
+  return (
+    <>
+      <DropZone
+        isActive={viewState.isDraggingDroppingFile}
+        onDrop={handleDrop}
+        onDragEnter={handleDragEnter}
+        onDragOver={handleDragOver}
+        onDragLeave={handleDragLeave}
+        onMouseLeave={handleMouseLeave}
       >
-        {this.props.viewState.isDraggingDroppingFile ? (
-          <div className={Styles.inner}>
+        {viewState.isDraggingDroppingFile ? (
+          <Box centered fullWidth fullHeight column>
+            <StyledIcon glyph={Icon.GLYPHS.dragDrop} styledHeight="80px" />
             <Trans i18nKey="dragDrop.text">
-              <h3 className={Styles.heading}>Drag & Drop</h3>
-              <div className={Styles.caption}>
+              <Text
+                textLight
+                textAlignCenter
+                pop
+                as="h3"
+                css={`
+                  margin-top: ${theme.spacing}px;
+                  font-weight: 700;
+                `}
+              >
+                Drag & Drop
+              </Text>
+              <Text textLight pop textAlignCenter styledFontSize="24px">
                 Your data anywhere to view on the map
-              </div>
+              </Text>
             </Trans>
-          </div>
-        ) : (
-          ""
-        )}
-      </div>
-    );
-  }
-}
-
-export default withTranslation()(withViewState(DragDropFile));
+          </Box>
+        ) : null}
+      </DropZone>
+      <DragDropNotification uploadedFiles={lastUploadedFiles} />
```

Comment:
Ideally we would render this only if `lastUploadedFiles` is not empty but I guess here we need it so that the CSS slide in transition works.

Reply:
Yeah, we need it for the styling